pkg/orchestrator/orchestrator.go

```diff
@@ -361,6 +370,124 @@ func (s *ControllerServer) permitListRuleAdd(c *gin.Context) {
 	}
 }
 
+// Add permit list rules to all resources within a tag
+func (s *ControllerServer) permitListRuleAddTag(c *gin.Context) {
+	tag := c.Param("tag")
+
+	// Parse permit list rules to add
+	var rule *paragliderpb.PermitListRule
+	if err := c.BindJSON(&rule); err != nil {
+		c.AbortWithStatusJSON(400, createErrorResponse(err.Error()))
+		return
+	}
+
+	// Resolve the tag to URIs
+	conn, err := grpc.NewClient(s.localTagService, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		c.AbortWithStatusJSON(400, createErrorResponse(err.Error()))
+		return
+	}
+	defer conn.Close()
+
+	// Send RPC to resolve tag
+	client := tagservicepb.NewTagServiceClient(conn)
+	resolvedTag, err := client.ResolveTag(context.Background(), &tagservicepb.ResolveTagRequest{TagName: tag})
+	if err != nil {
+		c.AbortWithStatusJSON(400, createErrorResponse(err.Error()))
+		return
+	}
+
+	// Add rule to each URI in the resolved tag
+	for _, mapping := range resolvedTag.Tags {
+		// Get the cloud and namespace from the tag
+		namespace, cloud, _, err := parseTag(mapping.Name)
+		if err != nil {
+			c.AbortWithStatusJSON(400, createErrorResponse(err.Error()))
+			return
+		}
+
+		// Create connection to cloud plugin
+		cloudClient, ok := s.pluginAddresses[cloud]
+		if !ok {
+			c.AbortWithStatusJSON(400, createErrorResponse("invalid cloud name"))
+			return
+		}
+		conn, err := grpc.NewClient(cloudClient, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
The grpc new client and initialization could be moved out of the loop

Reply:
But the tags may point to resources in different clouds, so we can't quite do that. I guess I could keep track of which clients have been initialized in the loop and use them if they have - not sure if that is worth it though. Thoughts?